common/v2/features/SendAssets/components/TransactionReceipt.tsx

```diff
@@ -37,15 +38,19 @@ export default function TransactionReceipt({ txReceipt, txConfig }: IStepCompone
     );
     if (blockNumber === 0 && txReceipt.hash) {
       const blockNumInterval = setInterval(() => {
-        getTransactionReceiptFromHash(txReceipt.hash, provider).then(transactionReceipt => {
-          if (transactionReceipt) {
-            const transactionStatus = transactionReceipt.status === 1 ? true : false;
+        getTransactionReceiptFromHash(txReceipt.hash, provider).then(transactionOutcome => {
+          if (transactionOutcome) {
```

Comment:
1. Can `getTransactionReceiptFromHash` really resolve to `undefined`?
Normally a resolved promise will always return a value, therefore this conditional is unnecessary.
Otherwise to reduce the indentation level prefer the defensive form:
```
if (!transactionOutcome) return;
```

2. Since it's orthogonal to the second condition i.e `if (timestamp === 0 && blockNumber !== 0)`
It would be cleaner to have two separate `effects`